src/main/java/org/apache/sling/junit/tests/TestReferenceJTest.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.sling.junit.tests;
+
+import org.apache.sling.junit.annotations.SlingAnnotationsTestRunner;
+import org.apache.sling.junit.annotations.TestReference;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static junit.framework.TestCase.assertEquals;
+import static junit.framework.TestCase.assertNotNull;
+import static junit.framework.TestCase.assertNull;
+
+
+@RunWith(SlingAnnotationsTestRunner.class)
+public class TestReferenceJTest {
```

Comment:
I'm not sure I am terribly fond of shipping Sling JUnit Core with test classes. At the very least we should make sure these are not added to the API (I think they aren't, but didn't verify). I would prefer if the test classes coould be moved to `src/test/**` and instead be built into an ad-hoc tinybundle for testing with PaxExam.

Reply:
I do understand this concern. It is reasonable. I spent time to trying to figure out how to run tests for the runtime annotations, but unfortunately, I don't get the big picture or the details about that would work. I guess one thing I can offer is to remove the test and test content at the end once everything else is resolved.